osgp/protocol-adapter-mqtt/osgp-protocol-adapter-mqtt/src/test/java/org/opensmartgridplatform/adapter/protocol/mqtt/application/services/SubscriptionServiceTest.java

```diff
@@ -53,21 +55,33 @@
   @Mock private MqttClientAdapter mqttClientAdapter;
   @Captor private ArgumentCaptor<ProtocolResponseMessage> protocolResponseMessageCaptor;
 
+  @Mock private Properties mqttClientProperties;
+
   @BeforeEach
   public void setUp() {
+
     this.instance =
         new SubscriptionService(
             this.mqttDeviceRepository,
             this.mqttClientAdapterFactory,
             this.outboundOsgpCoreResponseMessageSender,
-            DEFAULT_PORT,
-            DEFAULT_TOPICS,
-            DEFAULT_QOS.name());
+            this.mqttClientProperties);
+
     lenient().when(this.mqttClientAdapter.getMessageMetadata()).thenReturn(this.messageMetadata);
+
+    lenient()
+        .when(this.mqttClientProperties.getProperty(MqttConstants.DEFAULT_PORT_PROPERTY_NAME))
+        .thenReturn(String.valueOf(DEFAULT_PORT));
+    lenient()
+        .when(this.mqttClientProperties.getProperty(MqttConstants.DEFAULT_QOS_PROPERTY_NAME))
+        .thenReturn(DEFAULT_QOS.name());
+    lenient()
+        .when(this.mqttClientProperties.getProperty(MqttConstants.DEFAULT_TOPICS_PROPERTY_NAME))
+        .thenReturn(DEFAULT_TOPICS);
```

Comment:
As long as `Properties` are used, as far as the test results go, this works alright. But in a case like this wouldn't just setting up the actual properties  lead to simpler code than mocking the expected return values?

Reply:
Refactored.